### PR TITLE
Update Reader icons to Gridicons and universal vector assets.

### PR DESCRIPTION
**Pull request by mattmiklic**
Updates icons used in the reader to Gridicons and universal vector PDF assets.

Removed, replaced with `gravatar` (default user image):
- default-identicon
- gravatar-reader

Removed, replaced with `post-blavatar-placeholder` (default site icon):
- icon-feed

Updated:
- background-comment-nesting
- background-reader-tear
- gravatar
- icon-clear-textfield
- icon-posts-share
- icon-reader-comment-highlight
- icon-reader-comment-like
- icon-reader-comment-liked
- icon-reader-comment-reply
- icon-reader-comment
- icon-reader-follow-highlight
- icon-reader-follow
- icon-reader-following
- icon-reader-header-list
- icon-reader-header-tag
- icon-reader-like-highlight
- icon-reader-like
- icon-reader-liked
- icon-reader-search-plus

`background-comment-nesting.png` included a background color that doesn't match the background color of the app (see screenshot below). Its replacement, `background-comment-nesting.pdf` uses a transparent background instead. Let me know if anyone knows why/if we shouldn't use a transparent background for the comment nesting image.

background-comment-nesting-color

*Reader Comments*, before (left) and after (right). Includes `gravatar` replacing `default-identicon`; updated `icon-reader-comment-like`, `icon-reader-comment-liked`, `icon-reader-comment-reply`, `background-comment-nesting`.
comments-before-after

Updated to use Gridicons:

`icon-posts-share` 
share

`icon-clear-textfield`, `icon-reader-search-plus`
search-close

All other icons have been updated to vector PDF assets but are visually the same as before.

Needs Review: @aerych

### CODE REVIEW

**Comment by aerych** [comment]
Nice @mattmiklic !
>  Let me know if anyone knows why/if we shouldn't use a transparent background for the comment nesting image.

Transparencies degrade scrolling performance so we want to avoid them for those use cases.

**Comment by aerych** [comment]
This PR will close #4433

File: WordPress/Classes/ViewRelated/Reader/FollowedSitesViewController.m [hunk 1]
```diff
@@ -143,7 +143,7 @@ - (NSFetchRequest *)fetchRequest
 
 - (void)configureCell:(UITableViewCell *)cell atIndexPath:(NSIndexPath *)indexPath
 {
-    UIImage *defaultImage = [UIImage imageNamed:@"icon-feed"];
+    UIImage *defaultImage = [UIImage imageNamed:@"post-blavatar-placeholder"];
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Reader/FollowedSitesViewController.m]: Good call.

File: WordPress/Resources/AppImages.xcassets/icon-reader-comment.imageset/Contents.json [hunk 2]
```diff
@@ -2,18 +2,7 @@
...
-    {
-      "idiom" : "universal",
-      "scale" : "2x",
-      "filename" : "icon-reader-comment@2x.png"
-    },
-    {
-      "idiom" : "universal",
-      "scale" : "3x",
-      "filename" : "icon-reader-comment@3x.png"
+      "filename" : "icon-reader-comment-highlighted.pdf"
```
Comment (aerych) [inline WordPress/Resources/AppImages.xcassets/icon-reader-comment.imageset/Contents.json]: Naming mixmatch.

File: WordPress/Resources/AppImages.xcassets/icon-reader-comment-highlight.imageset/Contents.json [hunk 3]
```diff
@@ -2,18 +2,7 @@
...
-    {
-      "idiom" : "universal",
-      "scale" : "2x",
-      "filename" : "icon-reader-comment-highlight@2x.png"
-    },
-    {
-      "idiom" : "universal",
-      "scale" : "3x",
-      "filename" : "icon-reader-comment-highlight@3x.png"
+      "filename" : "icon-reader-comment.pdf"
```
Comment (aerych) [inline WordPress/Resources/AppImages.xcassets/icon-reader-comment-highlight.imageset/Contents.json]: Naming mixmatch

**Comment by aerych** [comment]
Hmm. Looks like [icon-reader-comment-highlighted.pdf](https://github.com/wordpress-mobile/WordPress-iOS/blob/update/reader-icons/WordPress/Resources/AppImages.xcassets/icon-reader-comment.imageset/icon-reader-comment-highlighted.pdf) and [icon-reader-comment.pdf](https://github.com/wordpress-mobile/WordPress-iOS/blob/update/reader-icons/WordPress/Resources/AppImages.xcassets/icon-reader-comment-highlight.imageset/icon-reader-comment.pdf) got mismatched.  But the way they were assigned in code was also reversed which canceled out the issue. :) 
Could we swap things around so the naming matches the use?

File: WordPress/Classes/ViewRelated/Reader/FollowedSitesViewController.m (hunk 1 above)
Comment (mattmiklic) [inline WordPress/Classes/ViewRelated/Reader/FollowedSitesViewController.m]: Thanks -- I forgot to include a before and after of this in the original post:

feedicons

**Comment by mattmiklic** [comment]
> Transparencies degrade scrolling performance so we want to avoid them for those use cases.

Thanks -- I thought there might be something like that at play. In 1fa67f8 I've updated `background-comment-nesting.pdf` to include a `greyLighten30` background so it matches the page below and still solves #4433.

**Comment by mattmiklic** [comment]
> Hmm. Looks like icon-reader-comment-highlighted.pdf and icon-reader-comment.pdf got mismatched. But the way they were assigned in code was also reversed which canceled out the issue. :) 
Could we swap things around so the naming matches the use?

Whoops! That was strange. Looks like I mis-named the assets but dropped them into the right place in Xcode. :) Fixed up in 997fee8.

**Comment by aerych** [comment]
Awesome. Took things for a test drive and they look great!  One thing that caught my eye (and this was already the case before, doH!) was that we don't have a jazzy orange version of the reply icon for its highlight state.  When tapped it does this: 
simulator screen shot jan 19 2016 2 47 51 pm
Not sure if that's something to tackle here or in a separate PR. If we want to wait then this one is ready to :shipit:

**Comment by mattmiklic** [comment]
> Not sure if that's something to tackle here or in a separate PR

Good catch, I noticed a handful of places like that while I was replacing icons. I think it would make sense to open an new issue to look at highlighted states in a cohesive way around the app -- I noticed that in some places we use a special -highlighted image, and in others we simply darken the existing one, and it would be nice to make that consistent everywhere.

**Pull request merged**